qemu/tests/win_irq_check.py

```diff
@@ -0,0 +1,56 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest import utils_test
+
+
+# This decorator makes the test function aware of context strings
+@error_context.context_aware
+def run(test, params, env):
+    """
+    QEMU windows guest vitio device irq check test
+
+    1) Start guest with virtio device.
+    2) Make sure driver verifier enabled in guest.
+    3) Get irq info in guest.
+    4) Check the value of irq number.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    def irq_check(session, device_name):
+        """
+        Return if virtio device's irq number is positive or not
+        :param session: use for sending cmd
+        :param device_name: virtio device's name
+        """
+        msinfo_cmd = params["msinfo_cmd"]
+        if session.cmd_status(msinfo_cmd, timeout=360):
+            test.error("Run msinfo32 failed.")
+        status, irq_dev_info = session.cmd_status_output(params["irq_cmd"]
+                                                         % device_name)
+        if status:
+            test.fail("Can't get %s's irq info." % device_name)
+        irq_value = re.split('\s+', irq_dev_info)[1]
+        logging.info("irq number is %s" % irq_value)
+        return int(irq_value)
+
+    driver = params["driver_name"]
+    device_name = params["device_name"]
+    timeout = int(params.get("login_timeout", 360))
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    error_context.context("Boot guest with %s device" % driver, logging.info)
+    utils_test.qemu.setup_win_driver_verifier(driver, vm, timeout)
```

Comment:
Before driver verifier check step,it's better to check whether driver is installed.
Could you create a function in utils_test to check required driver is installed?

Reply:
Yes,it's needed. But it's better to write another patch to track it, as not only this case but also the other cases need it, such as driver_in_use.py, qemu_guest_agent.py, rng_bat.py and so on.